src/main/java/juja/microservices/gamification/service/AchievementService.java

```diff
@@ -58,55 +57,76 @@ public String addDaily(DailyRequest request) {
             return achievementRepository.addAchievement(achievement);
         }
     }
 
     public List<String> addThanks(ThanksRequest request) {
-        requestValidator.checkThanks(request);
-
         String userFromId = request.getFrom();
         String userToId = request.getTo();
         String description = request.getDescription();
+
+        if (userFromId.equalsIgnoreCase(userToId)) {
+            logger.warn("User '{}' trying to put 'Thanks' achievement to yourself", request.getTo());
+            throw new UnsupportedAchievementException("You cannot thank yourself");
+        }
+
         List<Achievement> userFromAndToListToday = achievementRepository
                 .getAllAchievementsByUserFromIdCurrentDateType(userFromId, AchievementType.THANKS);
 
         for (Achievement achievement : userFromAndToListToday) {
-            if (achievement.getTo().equals(userToId)) {
-                logger.warn("User '{}' tried to give 'Thanks' achievement more than one times to person '{}'", userFromId, userToId);
-                throw new UnsupportedAchievementException("You cannot give more than one thanks for day one person");
-            }
+            checkTryToThanksYourself(userFromId, userToId, achievement);
         }
 
-        List<String> result = new ArrayList<>();
         if (userFromAndToListToday.isEmpty()) {
-            Achievement firstThanks = new Achievement(userFromId, userToId, 1, description, AchievementType.THANKS);
-            result.add(achievementRepository.addAchievement(firstThanks));
-            logger.info("Added first 'Thanks' achievement from user '{}' to user '{}'", userFromId, userToId);
-            return result;
+            return addFirstThanks(userFromId, userToId, description);
 
         } else if (userFromAndToListToday.size() >= TWO_THANKS) {
             logger.warn("User '{}' tried to give 'Thanks' achievement more than two times per day", userFromId);
             throw new UnsupportedAchievementException("You cannot give more than two thanks for day");
 
         } else {
-            Achievement secondThanks = new Achievement(userFromId, userToId, 1, description, AchievementType.THANKS);
-            result.add(achievementRepository.addAchievement(secondThanks));
-            logger.info("Added second 'Thanks' achievement from user '{}' to user '{}'", userFromId, userToId);
-            String descriptionTwoThanks = "Issued two thanks";
-            Achievement thirdThanks = new Achievement(userFromId, userFromId, 1, descriptionTwoThanks, AchievementType.THANKS);
-            result.add(achievementRepository.addAchievement(thirdThanks));
-            logger.info("Added 'Thanks' achievement to user '{}' for the distributed two thanks to other users", userFromId);
+            return addSecondThanks(userFromId, userToId, description);
         }
+    }
+
+    private void checkTryToThanksYourself(String userFromId, String userToId, Achievement achievement) {
```

Comment:
I think there is a mistake in the method's name. You check that this thanks should be the first one today. And the checking 'thanks to yourself' you do with 'if' condition above.